media/js/firefox/concerts.js

```diff
@@ -0,0 +1,84 @@
+(function(Mozilla) {
+    'use strict';
+
+    var stateStorageKey = 'fxaOauthState';
+    var verifiedStorageKey = 'fxaOauthVerified';
+    var cookieDays = 1;
+
+    var className;
+    var state;
+
+    /*
+        This function initiates the OAuth process.
+
+        It looks for a state key stored in a cookie, and, if not found, generates a new
+        key and places it in a cookie.
+
+        The state value is then set in the form and on the direct link URL.
+
+        Next, we retrieve metrics information from an asynchronous FxA call. When this
+        call completes, the submit button on the form is enabled.
+    */
+    function initOauth() {
+        var fxaFormWrapper = document.getElementById('fxa-form-wrapper');
+        var metricsFlowEndpoint = fxaFormWrapper.getAttribute('data-fxa-metrics-endpoint');
+        var flowIdField = document.getElementById('flow_id');
+        var flowBeginTimeField = document.getElementById('flow_begin_time');
+        var stateField = document.getElementById('state');
+        var submit = document.getElementById('submit');
+        var fxaSignIn = document.getElementById('fxa-sign-in');
+
+        // check for an existing state value in a cookie
+        if (Mozilla.Cookies.hasItem(stateStorageKey)) {
+            state = Mozilla.Cookies.getItem(stateStorageKey);
+        } else {
+            // generate a state token and put it in a cookie
+
+            // the suffix below is superfluous for now, but will help us identify
+            // different OAuth flows in the future
+            state = Math.random() + '_concert_q42018';
+
+            var d = new Date();
+            d.setTime(d.getTime() + (cookieDays * 24 * 60 * 60 * 1000));
+            Mozilla.Cookies.setItem(stateStorageKey, state, d.toUTCString(), '/');
```

Comment:
We should probably be using a [signed cookie](https://docs.djangoproject.com/en/1.8/topics/signing/) here, but it's not a blocker for this use case.